### ISSUE TITLE
REFACTOR DOUBLE LINKED LIST SWAP FUNCTION TO BE ABLE TO READ IT

### ISSUE THREAD

**Issue by TheOnlyMonster** [Nour8317/assignment_2_datastructure#1]
THE DOUBLE LINKED LIST FILE IS BECOMING DIFFICULT TO READ AND MAINTAIN, WITH LONG AND COMPLEX FUNCTIONS SPECIFICALLY SWAP FUNCTION, INCONSISTENT NAMING CONVENTIONS, AND NO COMMENTS. THIS MAKES IT HARD FOR ME TO UNDERSTAND AND MODIFY THE CODE. 
PLEASE **YASSIN TARK HELMY** MAKE THE CODE MORE READABLE SO I CAN GET TO KNOW WHAT THE FUNCTION DO AND PLEASE REMOVE ANY UNNECESSARY IF STATEMENTS

**Comment by abdelrahmannreda** [comment]
Agreed. Please make the necessary improvements to enhance the readability of the double linked list file, including refactoring long and complex functions like the swap function, standardizing naming conventions, adding comments for better understanding, and removing unnecessary if statements. Thank you

**Comment by YassinTarekHelmy** [comment]
> THE DOUBLE LINKED LIST FILE IS BECOMING DIFFICULT TO READ AND MAINTAIN, WITH LONG AND COMPLEX FUNCTIONS SPECIFICALLY SWAP FUNCTION, INCONSISTENT NAMING CONVENTIONS, AND NO COMMENTS. THIS MAKES IT HARD FOR ME TO UNDERSTAND AND MODIFY THE CODE. PLEASE **YASSIN TARK HELMY** MAKE THE CODE MORE READABLE SO I CAN GET TO KNOW WHAT THE FUNCTION DO AND PLEASE REMOVE ANY UNNECESSARY IF STATEMENTS

NO one cares

**Comment by TheOnlyMonster** [comment]
> > THE DOUBLE LINKED LIST FILE IS BECOMING DIFFICULT TO READ AND MAINTAIN, WITH LONG AND COMPLEX FUNCTIONS SPECIFICALLY SWAP FUNCTION, INCONSISTENT NAMING CONVENTIONS, AND NO COMMENTS. THIS MAKES IT HARD FOR ME TO UNDERSTAND AND MODIFY THE CODE. PLEASE **YASSIN TARK HELMY** MAKE THE CODE MORE READABLE SO I CAN GET TO KNOW WHAT THE FUNCTION DO AND PLEASE REMOVE ANY UNNECESSARY IF STATEMENTS
> 
> NO one cares

SHAME ON U

**Comment by abdelrahmannreda** [comment]
> > THE DOUBLE LINKED LIST FILE IS BECOMING DIFFICULT TO READ AND MAINTAIN, WITH LONG AND COMPLEX FUNCTIONS SPECIFICALLY SWAP FUNCTION, INCONSISTENT NAMING CONVENTIONS, AND NO COMMENTS. THIS MAKES IT HARD FOR ME TO UNDERSTAND AND MODIFY THE CODE. PLEASE **YASSIN TARK HELMY** MAKE THE CODE MORE READABLE SO I CAN GET TO KNOW WHAT THE FUNCTION DO AND PLEASE REMOVE ANY UNNECESSARY IF STATEMENTS
> 
> NO one cares

You are a bad developer ..
